src/migration/MigrationExecutor.ts

```diff
@@ -247,6 +247,84 @@ export class MigrationExecutor {
         }
     }
 
+    async undoMigrationsUntil(options: { name: string }): Promise<Migration[] | void> {
+        const queryRunner = this.queryRunner || this.connection.createQueryRunner("master");
+
+        // create migrations table if its not created yet
+        await this.createMigrationsTableIfNotExist(queryRunner);
+
+        // get all migrations that are executed and saved in the database
+        const executedMigrations = await this.loadExecutedMigrations(queryRunner);
+
+        // get migration to which we should revert
+        let targetMigration = executedMigrations.find(migration => migration.name === options.name);
+
+        // if no migration found in the database then nothing to revert
+        if (!targetMigration) {
+            this.connection.logger.logSchemaBuild(`No migration with name ${options.name} was found in the database. Nothing to revert!`);
+            return;
+        }
+
+
+        // get all user's migrations in the source code
+        const allMigrations = this.getMigrations();
+
+        const migrationsToRevert = executedMigrations
+            .filter(migration => migration.id! > targetMigration!.id!)
```

Comment:
Do we confirm the timestamps are always after one another & can make the assumption the ID order is the timestamp order?

Instead of ID this should be timestamp.  The problem is that the ID always goes up but if we revert and then add more in and then run code again, change values, etc, this may not be correct, will it?